src/plugins/data_source/server/client/configure_client.ts

```diff
@@ -29,53 +25,51 @@ export const configureClient = async (
   config: DataSourcePluginConfigType,
   logger: Logger
 ): Promise<Client> => {
-  const dataSource = await getDataSource(dataSourceId, savedObjects);
-  const rootClient = getRootClient(dataSource.attributes, config, openSearchClientPoolSetup);
+  try {
+    const dataSource = await getDataSource(dataSourceId, savedObjects);
+    const rootClient = getRootClient(dataSource.attributes, config, openSearchClientPoolSetup);
 
-  return getQueryClient(rootClient, dataSource, savedObjects, cryptographyClient);
+    return await getQueryClient(rootClient, dataSource, savedObjects, cryptographyClient);
+  } catch (error: any) {
+    logger.error(
+      `Fail to get data source client for dataSourceId: [${dataSourceId}]. Detail: ${error}`
+    );
+    // Re-throw as DataSourceError
+    throw new DataSourceError('Fail to get data source client: ', 400, error);
```

Comment:
not all error will be of 400? what if opensearch js client really failed with 500 issues?

Reply:
`configure_client.ts` has only the logic to create a client, nothing runtime. As far as I know, a Client object can be created with the wrong endpoint, wrong auth, no matter what. The runtime error will happen if caller use that client to make calls, E.g `client.search()`. Then that error is handled by the route handler directly. I don't see anything here that can be a 5xx

Reply:
what if the get saveobject call failed?

Reply:
then it creates a hapi/Boom error object, that's why you see in `DataSourceError` I add a conversion. You can take another look at the issue 2210 with the analysis, that why route handler throw 500